logstash-core/src/main/java/org/logstash/execution/AbstractPipelineExt.java

```diff
@@ -161,7 +166,7 @@ public class AbstractPipelineExt extends RubyBasicObject {
 
     private QueueReadClientBase filterQueueClient;
 
-    private ArrayList<FlowMetric> flowMetrics = new ArrayList<>();
+    private transient final ScopedFlowMetrics scopedFlowMetrics = new ScopedFlowMetrics();
```

Comment:
I didn't catch the intention (we are not serializing anywhere) of being `scopedFlowMetrics` but it doesn't break or hurt.

Reply:
I needed a way to provide the `PipelineIndicator.FlowObservation` with flow metrics that are _only_ for the pipeline itself, while still having a way to trigger them all to get captured, so this was an easy-enough way to provide both access patterns.